### PR TITLE
Fallback on ATen operator on ONNX assertion error

### LINKED ISSUE #17377: Exporting AdaptiveAvgPool2d to ONNX with ATen fallback produces error
## 🐛 Bug

When trying to export the AdaptiveAvgPool2d node to ONNX with  `OperatorExportTypes.ONNX_ATEN_FALLBACK`, the export fails, due to an assertion error being thrown.

## To Reproduce

1. Run the following code

```python
import torch
import torch.nn as nn
from torch.onnx import OperatorExportTypes


class AvgPool2dModule(nn.Module):
    def __init__(self, output_size):
        super().__init__()
        self.aap2d = nn.AdaptiveAvgPool2d(output_size=(output_size, output_size))
    def forward(self, inp):
        return self.aap2d(inp)
        
dummy_input = torch.randn(1, 3, 200, 400)
simple_module = AvgPool2dModule(2)

torch.onnx.export(simple_module, 
                  dummy_input,
                  'simple_module.onnx',
                  verbose=True,
                  operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK)
```

The stack trace:
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-4-91e8bc06839f> in <module>
     18                   'simple_module.onnx',
     19                   verbose=True,
---> 20                   operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK)
     21 

/usr/local/lib/python3.7/site-packages/torch/onnx/__init__.py in export(*args, **kwargs)
     25 def export(*args, **kwargs):
     26     from torch.onnx import utils
---> 27     return utils.export(*args, **kwargs)
     28 
     29 

/usr/local/lib/python3.7/site-packages/torch/onnx/utils.py in export(model, args, f, export_params, verbose, training, input_names, output_names, aten, export_raw_ir, operator_export_type)
    102             operator_export_type = OperatorExportTypes.ONNX
    103     _export(model, args, f, export_params, verbose, training, input_names, output_names,
--> 104             operator_export_type=operator_export_type)
    105 
    106 

/usr/local/lib/python3.7/site-packages/torch/onnx/utils.py in _export(model, args, f, export_params, verbose, training, input_names, output_names, operator_export_type, export_type, example_outputs, propagate)
    279                                                training, input_names,
    280                                                output_names, operator_export_type,
--> 281                                                example_outputs, propagate)
    282 
    283     # TODO: Don't allocate a in-memory string for the protobuf

/usr/local/lib/python3.7/site-packages/torch/onnx/utils.py in _model_to_graph(model, args, f, verbose, training, input_names, output_names, operator_export_type, example_outputs, propagate)
    225         params = list(_unique_state_dict(model).values())
    226 
--> 227     graph = _optimize_graph(graph, operator_export_type)
    228 
    229     # NB: ONNX requires complete information about output types, which might be

/usr/local/lib/python3.7/site-packages/torch/onnx/utils.py in _optimize_graph(graph, operator_export_type)
    153 
    154     if operator_export_type != OperatorExportTypes.RAW:
--> 155         graph = torch._C._jit_pass_onnx(graph, operator_export_type)
    156         torch._C._jit_pass_lint(graph)
    157         torch._C._jit_pass_onnx_peephole(graph)

/usr/local/lib/python3.7/site-packages/torch/onnx/__init__.py in _run_symbolic_function(*args, **kwargs)
     50 def _run_symbolic_function(*args, **kwargs):
     51     from torch.onnx import utils
---> 52     return utils._run_symbolic_function(*args, **kwargs)
     53 
     54 

/usr/local/lib/python3.7/site-packages/torch/onnx/utils.py in _run_symbolic_function(g, n, inputs, env, operator_export_type)
    502                     return None
    503                 fn = getattr(torch.onnx.symbolic, op_name)
--> 504                 return fn(g, *inputs, **attrs)
    505 
    506         elif ns == "prim":

/usr/local/lib/python3.7/site-packages/torch/onnx/symbolic.py in wrapper(g, *args)
     87             assert len(arg_descriptors) == len(args)
     88             args = [_parse_arg(arg, arg_desc) for arg, arg_desc in zip(args, arg_descriptors)]
---> 89             return fn(g, *args)
     90         # In Python 2 functools.wraps chokes on partially applied functions, so we need this as a workaround
     91         try:

/usr/local/lib/python3.7/site-packages/torch/onnx/symbolic.py in adaptive_avg_pool2d(g, input, output_size)
    598 @parse_args('v', 'is')
    599 def adaptive_avg_pool2d(g, input, output_size):
--> 600     assert output_size == [1, 1], "Only output_size=[1, 1] is supported"
    601     return g.op("GlobalAveragePool", input)
    602 

AssertionError: Only output_size=[1, 1] is supported
```
## Expected behavior

The export falls back to exporting it as an ATen node, as it was not able to properly export it as an ONNX node.

The verbose output expected from the snipped above:
```
graph(%input : Float(1, 3, 200, 400)) {
  %1 : int[] = onnx::Constant[value= 2  2 [ CPULongType{2} ]]()
  %2 : Float(1, 3, 2, 2) = aten::ATen[operator="adaptive_avg_pool2d"](%input, %1), scope: AvgPool2dModule/AdaptiveAvgPool2d[aap2d]
  return (%2);
}
```

## Environment

```
Collecting environment information...
PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.3
GCC version: Could not collect
CMake version: version 3.13.3

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy==1.16.1
[pip3] torch==1.0.1.post2
[conda] Could not collect
```

### PR DESCRIPTION
**Pull request by maximegregoire**
This ensures that when exporting nodes to ONNX with OperatorExportTypes.ONNX_ATEN_FALLBACK, it correctly falls back on ATen operator export when an AssertionError is raised from the regular export.

Fixes #17377

cc @BowenBao @neginraoof

### CODE REVIEW

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -506,7 +509,15 @@ def _run_symbolic_function(g, n, inputs, env, operator_export_type=OperatorExpor
...
                     return None
                 fn = getattr(torch.onnx.symbolic, op_name)
-                return fn(g, *inputs, **attrs)
+                try:
+                    return fn(g, *inputs, **attrs)
+                except AssertionError:
+                    if is_aten_fallback_export:
+                        # Falling back to ATen
+                        warnings.warn("An assertion error occured when trying to export {}. Falling back to ATen"
+                                      .format(op_name))
```
Comment (VitalyFedyunin) [inline torch/onnx/utils.py]: Makes sense to add error's text as well.

Comment (maximegregoire) [reply torch/onnx/utils.py]: You're right. The text has been added.

**Comment by garymm** [comment]
This is in theory still useful, even though the particular export error it addresses was already fixed.
@maximegregoire if you'd like to get this merged, please rebase it, add a unit test, and I'll review it.

**Comment by garymm** [comment]
Closing since this seems to have been abandoned by @maximegregoire. If you want to revive this, please rebase it and add a unit test.

**Pull request closed without merging**